snapcraft/internal/repo/_deb.py

```diff
@@ -186,20 +188,91 @@ def get_packages_for_source_type(cls, source_type):
         return packages
 
     @classmethod
-    def install_build_packages(cls, package_names):
-        unique_packages = set(package_names)
+    def _setup_multi_arch(cls, apt_cache, package_name):
+        if ':' not in package_name or package_name.endswith(':native'):
+            return False
+
+        (name, arch) = package_name.split(':', 2)
+        foreign_archs = subprocess.check_output(
+            ['dpkg', '--print-foreign-architectures']).decode()
+        if arch not in foreign_archs:
+            logger.info('Adding foreign architecture {}'.format(arch))
+            subprocess.check_call(
+                ['sudo', 'dpkg', '--add-architecture', arch])
+
+        sources = _get_local_sources_list()
+        if '[arch={}]'.format(arch) not in sources:
+            logger.info('Adding sources for {}'.format(arch))
+            release = platform.linux_distribution()[2]
+            sources = _format_sources_list(None, deb_arch=arch,
+                                           release=release, foreign=True)
+            with tempfile.NamedTemporaryFile() as sources_arch:
```

Comment:
can you rename `source_arch` to source_arch_file`? makes the code below a faster read